javascript/javascript2/week1/script.js

```diff
@@ -0,0 +1,90 @@
+// Shortest word
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+function compareLength(arr) {
+  return arr.reduce((x, y) => (x.length <= y.length ? x : y));
+}
+console.log("The shortest word in array is " + compareLength(danishWords));
+// returns 'ø' << Added console log here instead of consolelogging inside func
+
+//Danish letter amount
+function countLetters(string) {
+  console.log(string);
+  const result = { total: 0, æ: 0, ø: 0, å: 0 };
+  [...string].forEach((element) => {
+    element.match(/^[æøå]*$/) ? result.total++ : result.total;
+    element.match("æ") ? result.æ++ : result.æ;
+    element.match("ø") ? result.ø++ : result.ø;
+    element.match("å") ? result.å++ : result.å;
+  });
+  return console.log(result);
+}
+//thanQ, my brain almost melted constructing this :)
+
+const danishString = "Jeg har en blå bil";
+countLetters(danishString); // returns {total: 1, å: 1}
+/* Almost correct output, i am unsure how to increase object's properties without assigning them first, 
```

Comment:
There's several ways to check if a key exists in an object: 
 
```javascript
const x = {};
const myKey = "key";
const keyExistsMethodOne = myKey in x; // false 
const keyExistsMethodTwo = !x[myKey]; // true (ie. it is true that x[myKey] doesn't exist) 
``` 
 
In this way you could rewrite the above function to something like: 
```javascript 
function countDanishLetters(danishString){
    const danishLetters = ['æ', 'ø', 'å'];
    const tally = {};

    let total = 0;
    for(char of danishString) {
        if (danishLetters.includes(char)) {
            const charDoesntExistsInTally = !tally[char];
            tally[char] = charDoesntExistsInTally ? 1 : tally[char] + 1;
            total += 1;
        }
    }

    tally['total'] = total;
    return tally;
}
```